### ISSUE TITLE
DOC: update dissolve docstring to document options for by (e.g. list of attributes)

### ISSUE THREAD

**Issue by sehHeiden** [geopandas/geopandas#2895]
Currently dissolve has a parameter `by` which is restricted to a string. 
I think the option, that this parameter could also be a list of strings could would be great.  That way, all attributes in the list needs to be equal to combine the polygons.
Current workaround is to concat the values of the attributes into a new attribute and to delete it.

**Comment by martinfleis** [comment]
This already works. You can pass any array of a correct length to `by`. I see that the docstring says otherwise but it works. We should update the docstring to reflect the actual functionality.

```py
import geopandas
from geodatasets import get_path

nybb = geopandas.read_file(get_path("nybb"))
nybb.dissolve(by=["a", "a", "b", "b", "c"])
```

**Comment by anudeepadi** [comment]
Hey @martinfleis can I work on this as this seems like a good first-time issue for me to contribute. It seems I have to work on updating the dissolve function's docstring to accurately document the functionality of the by parameter. Let me know if there is anything more to do.

**Comment by martinfleis** [comment]
@anudeepadi That's it, we just need to ensure that a description of the parameter reflects its actual capabilities.

**Comment by anudeepadi** [comment]
Done Pl check #2961

**Comment by vidlb** [comment]
What is the current state of this ?  
I'm not sure to understand if I should downgrade pandas to < 1.3 ?  
I face this problem with both string and integer fields that aren't preserved during aggregation (a in range index is returned instead of the desired index or multi-index).  
Tested with geopandas 0.12 and 0.14, pandas 1.5 and 2.1

**Comment by martinfleis** [comment]
Closed by #2961

**Comment by anudeepadi** [comment]
@martinfleis I'm interested in contributing to this project. I have availability to work on open issues, particularly those labeled 'help wanted' or suitable for new contributors. Please keep me in mind if any issues come up that you think I could assist with. I'm eager to help improve this software, and happy to take direction from you and the other maintainers on what would be most useful. Just let me know how I can best contribute my skills and time. Looking forward to collaborating with the community!
